### PR TITLE
Refactor and Overhaul

### PR DESCRIPTION
**Pull request by jsquire**
Refactoring to remove Event Hubs + Fabric Bot integration and revise how initialization is performed.  Going forward, only a single copy of each model will be loaded.  Model initialization is now performed in the body of the prediction service; the prior approach to initializing in the constructor made it more difficult to track failures, as there was no corresponding Http Request tied to it.

### CODE REVIEW

File: src/IssueLabeler/IssueLabeler.csproj [hunk 1]
```diff
@@ -4,6 +4,11 @@
     <AzureFunctionsVersion>v4</AzureFunctionsVersion>
     <UserSecretsId>401bcdfb-b94b-448f-80ac-36ea8a447a97</UserSecretsId>
   </PropertyGroup>
+  <ItemGroup>
+    <Compile Remove="Models\**" />
+    <EmbeddedResource Remove="Models\**" />
+    <None Remove="Models\**" />
+  </ItemGroup>
```
Comment (jsquire) [inline src/IssueLabeler/IssueLabeler.csproj]:
```suggestion
```

Comment (jsquire) [reply src/IssueLabeler/IssueLabeler.csproj]: Not sure where this came from.

File: src/IssueLabeler/AzureSdkIssueLabelerService.cs [hunk 2]
```diff
@@ -40,6 +54,47 @@ public async Task<ActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "P
...
+                    }
+                }
+                catch (Exception ex)
+                {
+                    log.LogError($"Error initializing the label prediction models: {ex.Message}{Environment.NewLine}\t{ex}{Environment.NewLine}");
+                    return EmptyResult;
+                }
+                finally
+                {
+                    if (semaphoreHeld)
```
Comment (christothes) [inline src/IssueLabeler/AzureSdkIssueLabelerService.cs]: nit: I think you can replace the need for `semaphoreHeld` with a check like: 
```c#
if (InitSync.CurrentCount <= 0)
```

File: src/IssueLabeler/AzureSdkIssueLabelerService.cs [hunk 3]
```diff
@@ -51,23 +106,71 @@ public async Task<ActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "P
...
+            {
+                foreach (var repo in commonModelRepos.Split(';', StringSplitOptions.RemoveEmptyEntries))
+                {
+                    CommonModelRepositories.Add(repo);
+                }
+            }
+        }
+
+        private string TranslateRepoName(string repoName) =>
+            CommonModelRepositories.Contains(repoName)
```
Comment (christothes) [inline src/IssueLabeler/AzureSdkIssueLabelerService.cs:167]: is contains ever going to be a problem where a repo is a substring of another? Probably rare, but `-pr` repos are one example I can think of.

File: src/IssueLabeler/IssueLabeler.csproj (hunk 1 above)
Comment (christothes) [reply src/IssueLabeler/IssueLabeler.csproj]: VS is "helpful" this way when you delete stuff through the solution explorer.

**Review by christothes — APPROVED** [review]

File: src/IssueLabeler/AzureSdkIssueLabelerService.cs (hunk 2 above)
Comment (jsquire) [reply src/IssueLabeler/AzureSdkIssueLabelerService.cs]: huh...  I never realized that property was there.   😊

File: src/IssueLabeler/AzureSdkIssueLabelerService.cs (hunk 3 above)
Comment (jsquire) [reply src/IssueLabeler/AzureSdkIssueLabelerService.cs:167]: If we get that far, we'll probably want a full-blown map, as the "pr" variants would use the model from the public version of those repositories.

**Pull request merged**